packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.styles.tsx

```diff
@@ -0,0 +1,142 @@
+import {
+  FontSizes,
+  HighContrastSelector,
+  ScreenWidthMaxMedium,
+  ScreenWidthMaxSmall,
+  AnimationVariables,
+  normalize
+} from '../../Styling';
+import { SearchBoxBase } from './SearchBox.base';
+import { ISearchBoxProps, ISearchBoxStyleProps, ISearchBoxStyles } from './SearchBox.types';
+
+export function getStyles(props: ISearchBoxStyleProps): ISearchBoxStyles {
+  const { theme, underlined, disabled, hasFocus } = props;
+  const { palette, fonts } = theme;
+
+  return {
+    root: [
+      'ms-Searchbox',
+      fonts.medium,
+      normalize,
+      {
+        color: palette.neutralPrimary,
+        display: 'flex',
+        flexDirection: 'row',
+        flexWrap: 'nowrap',
+        alignItems: 'stretch',
+        padding: '0 0 0 8px',
+        border: `1px solid ${palette.neutralTertiary}`,
+        height: '32px',
+        selectors: {
+          [HighContrastSelector]: {
+            border: '1px solid WindowText'
+          },
+          ':hover': {
+            borderColor: palette.neutralDark,
+            $label: {
+              color: palette.black,
+              $iconContainer: {
+                color: palette.themeDark
+              }
+            }
+          }
+        }
+      },
+      hasFocus && {
+        borderColor: palette.themePrimary,
+        selectors: {
+          ':hover': {
+            borderColor: palette.themePrimary,
+          },
+          [HighContrastSelector]: {
+            borderColor: 'Highlight'
+          }
+        }
+      },
+      disabled && {
+        borderColor: palette.neutralLighter,
+        backgroundColor: palette.neutralLighter,
+        pointerEvents: 'none',
+        cursor: 'default',
+      },
+      underlined && {
+        borderWidth: '0 0 1px 0'
+      },
+      underlined && disabled && {
+        backgroundColor: 'transparent'
+      }
+    ],
+    iconContainer: [
+      'ms-Searchbox-iconContainer',
+      {
+        display: 'flex',
+        flexDirection: 'column',
+        justifyContent: 'center',
+        flexShrink: 0,
+        fontSize: 16,
+        width: 32,
+        textAlign: 'center',
+        transition: `width ${AnimationVariables.durationValue1}`,
+        color: palette.themePrimary
+      },
+      hasFocus && {
+        width: '4px',
+        transition: `width  ${AnimationVariables.durationValue1}`
+      },
+      disabled && {
+        color: palette.neutralTertiary
+      }
+    ],
+    icon: [
+      'ms-Searchbox-icon',
+      {
+        opacity: 1,
+        transition: `opacity ${AnimationVariables.durationValue1} 0s`
+      },
+      hasFocus && {
+        opacity: 0,
+        transition: `opacity 0 ${AnimationVariables.durationValue1}`
+      }
+    ],
+    clearButton: [
+      'ms-Searchbox-clearButton',
+      {
+        display: 'flex',
+        flexDirection: 'row',
+        alignItems: 'stretch',
+        cursor: 'pointer',
+        flexBasis: '32px',
```

Comment:
Not sure if flexBasis as a  number works, but the padding below should be a number instead of a string

Reply:
yeah, seems flexBasis needs a united value.